mono/utils/mono-threads.c

```diff
@@ -1144,6 +1144,12 @@ mono_thread_info_abort_socket_syscall_for_close (MonoNativeThreadId tid)
 	}
 
 	mono_thread_info_suspend_lock ();
+	/* lookup info again in case we raced with shutdown of the other thread and it's already gone.*/
+	info = mono_thread_info_lookup (tid);
+	if (!info) {
+		mono_thread_info_suspend_unlock ();
```

Comment:
Missing `mono_hazard_pointer_clear (hp, 1);` (in fact, also already at L1139)

Reply:
if info is null, isn't hazard pointer 1 already clear?

Reply:
Yeah that's what I understood afterwards :) We do have the pattern where we clear it anyway elsewhere: https://github.com/mono/mono/blob/master/mono/utils/mono-threads.c#L849